utility/ip.go

```diff
@@ -10,20 +10,29 @@ import (
 // X-Forwarded-For has a list of IPs, of which the first is the one of the original client.
 // This value however might not be necessarily trusted, as it can be forged by a malicious user.
 func ExtractIP(r *http.Request) string {
```

Comment:
I like your original suggestion to rename to `ExtractRemoteAddr`, especially since this is also called in the [logHandler](https://github.com/elastic/apm-server/blob/master/beater/handlers.go#L227) and in the [IP ratelimiter](https://github.com/elastic/apm-server/blob/master/beater/handlers.go#L266). 
Then you could move `net.ParseIP(s)` to `utility.ExtractIP` which internally uses `utility.ExtractRemoteAddr`.